saskatoon/harvest/views.py

```diff
@@ -277,3 +276,69 @@ def harvest_yield_create(request):
             messages.success(request, _("New Fruit Recipient successfully added!"))
 
         return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
+
+
+@login_required
+def harvest_adopt(request, id):
+    """
+    Adds harvest pickleader and changes status to Adopted
+    Used in a button at harvest-detail/status template
+    """
+    harvest = get_object_or_404(Harvest, id=id)
+    current_user = request.user
```

Comment:
Here we should make sure that user belongs to either the 'pickleader' or 'core' role groups, e.g.:
```
if request.user.groups.filter(name__in=["pickleader", "core"]).exists():
```